pkgs/development/libraries/libprotobuf-mutator/default.nix

```diff
@@ -0,0 +1,69 @@
+{ lib
+, clangStdenv
+, cmake
+, fetchFromGitHub
+, git
+, libtool
+, lzma
+, ninja
+, pkg-config
+, protobuf
+, zlib
+}:
+
+let
+  # See `src/cmake/external/googletest.cmake`.
+  googletest_src = fetchFromGitHub {
+    owner = "google";
+    repo = "googletest";
+    rev = "release-1.11.0";
+    sha256 = "sha256-SjlJxushfry13RGA7BCjYC9oZqV4z6x8dOiHfl/wpF0=";
+  };
+in
+clangStdenv.mkDerivation rec {
+  pname = "libprotobuf-mutator";
+  version = "1.0";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-E4EL0c/v1IAZ7nDQQFRflls5BG5juUfq0ztTBuAGDXk=";
+  };
+
+  patches = [ ./local_googletest.patch ];
+
+  postPatch = ''
+    # Remove examples as they require to pull external repositories.
+    rm -R examples/
+    sed -e 's/add_subdirectory(examples EXCLUDE_FROM_ALL)//' -i CMakeLists.txt
+
+    # Pulls an external git repository; Unused except in examples.
+    rm cmake/external/expat.cmake
+
+    cp -R ${googletest_src}/ googletest/
```

Comment:
Why not link?

Reply:
Because then the tests fail with a lot of errors `chmod: changing permissions of 'googletest/*': Operation not permitted`.

Reply:
Because it seems necessary to update the permissions on the files.